extension/goplugin/util.go

```diff
@@ -118,16 +199,54 @@ func resourceFromMap(context map[string]interface{}, rawType reflect.Type) (res
 					if field.Kind() == reflect.Int && value.Kind() == reflect.Float64 { // reflect treats number(N, 0) as float
 						field.SetInt(int64(value.Float()))
 					} else {
-						return nil, fmt.Errorf("invalid type of '%s' field (%s, expecting %s)", propertyName, value.Kind(), field.Kind())
+						return fmt.Errorf("invalid type of '%s' field (%s, expecting %s)", propertyName, value.Kind(), field.Kind())
 					}
 				}
 			}
 		}
 	}
 
-	return resource.Interface(), nil
+	return nil
 }
 
+func anyOfKinds(kind reflect.Kind, kinds []reflect.Kind) bool {
+	for _, k := range kinds {
```

Comment:
Use multicase switch for this, iteration is bad

Reply:
done for primitive and nullable types, I've left this method for short slices